lib/fetch/body.js

```diff
@@ -71,17 +64,61 @@ function extractBody (object, keepalive = false) {
 
     // Set source to a copy of the bytes held by object.
     source = new Uint8Array(object)
+  } else if (object instanceof FormData) {
+    const boundary = '----formdata-undici-' + Math.random()
+    const prefix = `--${boundary}\r\nContent-Disposition: form-data; name="`
+
+    /*! formdata-polyfill. MIT License. Jimmy Wärting <https://jimmy.warting.se/opensource> */
+    const escape = (str, filename) =>
+      (filename ? str : str.replace(/\r?\n|\r/g, '\r\n'))
+        .replace(/\n/g, '%0A')
+        .replace(/\r/g, '%0D')
+        .replace(/"/g, '%22')
+
+    // Set action to this step: run the multipart/form-data
+    // encoding algorithm, with object’s entry list and UTF-8.
+    action = async function * () {
+      for (const [name, value] of object) {
+        let str
+        if (typeof value === 'string') {
+          str =
+            prefix +
+            escape(name) +
+            `"\r\n\r\n${value.replace(/\r(?!\n)|(?<!\r)\n/g, '\r\n')}\r\n`
+        } else {
+          str =
+            (prefix +
+              escape(name) +
+              `"; filename="${escape(value.name, 1)}"\r\n` +
+              `Content-Type: ${
+                value.type || 'application/octet-stream'
+              }\r\n\r\n`,
+            await value.text(), // TODO: chunkify once we have Blob.stream()
+            '\r\n')
+        }
+
+        yield Buffer.from(str)
+      }
+
+      yield Buffer.from(`--${boundary}--`)
+    }
+
+    // Set source to object.
+    source = object
+
+    // Set length to unclear, see html/6424 for improving this.
+    // TODO
+
+    // Set Content-Type to `multipart/form-data; boundary=`,
+    // followed by the multipart/form-data boundary string generated
+    // by the multipart/form-data encoding algorithm.
+    contentType = 'multipart/form-data; boundary=' + boundary
   } else if (object instanceof Blob) {
     // Blob
 
     // Set action to this step: read object.
-    action = async (onNext, onError, onComplete) => {
-      try {
-        onNext(await object.arrayBuffer())
-        onComplete()
-      } catch (err) {
-        onError(err)
-      }
+    action = async function * () {
+      yield await object.arrayBuffer() // TODO: chunkify once we have Blob.stream()
```

Comment:
@jasnell: Any plans for `Blob.stream`?

Reply:
Blob.stream landed in `HEAD` last week :-)